Source/Light.swift

```diff
@@ -14,19 +14,20 @@ public struct Light: Equatable, CustomStringConvertible {
 	public let connected: Bool
 	public let group: Group?
 	public let location: Location?
+    public private(set) var touchedAt: NSDate?
```

Comment:
Ahhhh, this class is suppose to be immutable so all properties should be constants. That's why there is a `lightWithProperties` function that returns a new instance with the updated properties.

Secondly, you could default this to the UNIX epoch and then you could do away with the optional.